spec/uvarint_spec.cr

```diff
@@ -1,77 +1,191 @@
 require "spec"
 require "../src/uvarint"
 
-describe UVarint do
-  it "encodes and decodes random values" do
-    (0..100000).each do
-      n = Random.rand(0..0x7FFFFFFF).to_u64
-      e = UVarint.encode n
-      d = UVarint.decode e
-      d.should eq(n)
+describe UVarInt do
+  describe "instantiate" do
+    it "handles uint8" do
+      v = UVarInt.new 64_u8
+      b = v.bytes
+      b.should eq(Bytes[64])
+    end
+
+    it "handles uint16" do
+      v = UVarInt.new 300_u16
+      b = v.bytes
+      b.should eq(Bytes[172, 2])
+    end
+
+    it "handles uint32" do
+      v = UVarInt.new 70000_u32
+      b = v.bytes
+      b.should eq(Bytes[240, 162, 4])
+    end
+
+    it "handles uint64" do
+      v = UVarInt.new 5000000000_u64
+      b = v.bytes
+      b.should eq(Bytes[128, 228, 151, 208, 18])
+    end
+
+    it "handles arrays" do
+      a = [0xAC_u8, 0x02_u8]
+      v = UVarInt.new a
+      u = v.uint
+      u.should eq(300)
+    end
+
+    it "handles deques" do
+      d = Deque{0xAC_u8, 0x02_u8}
+      v = UVarInt.new d
+      u = v.uint
+      u.should eq(300)
+    end
+
+    it "handles static arrays" do
+      s = StaticArray[0xAC_u8, 0x02_u8]
+      v = UVarInt.new s
+      u = v.uint
+      u.should eq(300)
+    end
+
+    it "handles tuples" do
+      t = {0xAC_u8, 0x02_u8}
+      v = UVarInt.new t
+      u = v.uint
+      u.should eq(300)
+    end
+
+    it "handles ranges" do
+      r = 0x00_u8..0x09_u8
+      v = UVarInt.new r
+      u = v.uint
+      u.should eq(0)
+    end
+
+    it "handles strings" do
+      s = "abcd"
+      v = UVarInt.new s
+      u = v.uint
+      u.should eq(97)
     end
-  end
 
-  it "decodes single bytes" do
-    n = Random.rand(0..0x7F).to_u8
-    buf = [n]
-    d = UVarint.decode buf
-    d.should eq(n)
+    it "raises an exception if passed more than 10 bytes" do
+      b = Array.new(20, 0_u8)
+      ex = expect_raises(ArgumentError) { UVarInt.new b }
+      ex.message.should eq("cannot initialize with more than 10 bytes")
```

Comment:
Yeah, if BigInt is an option, let's use that. UVarint should be allowed to be arbitrarily large, otherwise the spec would just require Int64